src/DynamoCoreWpf/ViewModels/Core/GalleryViewModel.cs

```diff
@@ -104,6 +101,30 @@ public GalleryViewModel(DynamoViewModel dynamoViewModel)
         }
 
         /// <summary>
+        /// Set image path relative to the gallery Directory.
+        /// It looks for the images side-by-side with the gallerycontents.xml,
+        /// if it's can't be found, it looks one level up in gallery/Data
+        /// </summary>
+        /// <param name="galleryDirectory">
+        /// The directory in which the gallerycontents.xml resides
+        /// </param>
+        private void SetImagePath(string galleryDirectory)
+        {
+            foreach (GalleryContent content in contents)
+            {
+                var imagePath = Path.Combine(galleryDirectory, content.ImagePath);
+                if (File.Exists(imagePath))
+                {
+                    content.ImagePath = imagePath;
+                }
+                else
+                {
+                    content.ImagePath = Path.Combine(galleryDirectory, @"..\Data", content.ImagePath);
```

Comment:
Would it be possible to use `Path.GetFullPath` and resolve the absolute path instead of having `..` in the file path? Another question is, what happens when the image file cannot be found anywhere? Please make sure it does not crash Dynamo.